### PR TITLE
ViteJs

### PR DESCRIPTION
**Pull request by usu**
Revisiting #484 as I had to use again my (slow) notebook for the upcoming Hackathon :-)

Is this something that looks interesting to you?

I guess I could make it workable with another hour or two. At the moment only the Login screen and the home screen works.

Pending:

- [x] Fix font inter-ui
- [x] Fix all imports without file ending `.vue`
- [x] Find replacement for Webpack's require.context (https://github.com/vitejs/vite/issues/77)

### CODE REVIEW

**Comment by manuelmeister** [comment]
Can Vite only handle files with .vue extension?

**Comment by carlobeltrame** [comment]
Nice! Ideally I'd like to use the same build tool for local development and building for production, so there are no unexpected surprises. Can vite also build the finished bundle?
Also, would this work for the client side part of the print service as well? (not a big problem if it does not in my eyes, since it's a separate service)

**Comment by usu** [comment]
> Can Vite only handle files with .vue extension?

Vite should be able to handle anything that is Javascript. If I recall correctly, with version 2 Vite not even depends on Vue anymore and is framework agnostic.

What changes is that the file extension cannot be omitted anymore for importing Vue files.  This is by design and will stop working in upcoming releases of `vue-cli` as well.

https://github.com/vitejs/vite/issues/178#issuecomment-630138450

**Comment by usu** [comment]
> Nice! Ideally I'd like to use the same build tool for local development and building for production, so there are no unexpected surprises. Can vite also build the finished bundle?

Yes, `vite` can also build for production:
https://vitejs.dev/guide/build.html

It uses rollup under the hood for building. Haven't tried it yet, though.

**Comment by usu** [comment]
> Also, would this work for the client side part of the print service as well? (not a big problem if it does not in my eyes, since it's a separate service)

Theoretically, yes. Haven't tried it yet & it's still marked as experimental:
https://vite.nuxtjs.org/

File: frontend/src/components/form/base/BaseComponent.vue [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 <script>
 import { ValidationProvider } from 'vee-validate'
-import { formComponentPropsMixin } from '@/mixins/formComponentPropsMixin'
+import { formComponentPropsMixin } from '@/mixins/formComponentPropsMixin.s'
```
Comment (pmattmann) [inline frontend/src/components/form/base/BaseComponent.vue]: typo

**Review by pmattmann — APPROVED** [review]

**Review by manuelmeister — APPROVED** [review]
Cool!

File: frontend/package.json [hunk 2]
```diff
@@ -63,6 +64,10 @@
     "sass": "1.32.8",
     "sass-loader": "10.1.1",
     "source-map-loader": "1.1.3",
+    "vite": "^2.1.5",
```
Comment (BacLuc) [inline frontend/package.json]: ich würde hier gleich die version pinnen, sonst macht das renovate nachher

File: frontend/src/components/activity/ContentNode.vue [hunk 3]
```diff
@@ -9,12 +9,42 @@
...
+import ColumnLayout from './content/ColumnLayout.vue'
+import Notes from './content/Notes.vue'
+import Material from './content/Material.vue'
+import LAThematicArea from './content/LAThematicArea.vue'
+import SafetyConcept from './content/SafetyConcept.vue'
+import Storycontext from './content/Storycontext.vue'
+
+let contentNodeComponents = {}
+/*
+// Webpack is running (require.context needs to be available)
```
Comment (BacLuc) [inline frontend/src/components/activity/ContentNode.vue]: i would delete this code, we have git for versioning

File: frontend/src/plugins/formBaseComponents.js [hunk 4]
```diff
@@ -1,19 +1,41 @@
 // Global registration of all Vue components in folder components/form/base
 
-import upperFirst from 'lodash/upperFirst'
-import camelCase from 'lodash/camelCase'
+// import upperFirst from 'lodash/upperFirst'
```
Comment (BacLuc) [inline frontend/src/plugins/formBaseComponents.js]: i would delete commented code which is not needed anymore

File: frontend/src/plugins/formBaseComponents.js [hunk 5]
```diff
@@ -1,19 +1,41 @@
...
+import EColorPicker from '@/components/form/base/EColorPicker.vue'
+import EDatePicker from '@/components/form/base/EDatePicker.vue'
+import ERichtext from '@/components/form/base/ERichtext.vue'
+import ESelect from '@/components/form/base/ESelect.vue'
+import ESwitch from '@/components/form/base/ESwitch.vue'
+import ETextarea from '@/components/form/base/ETextarea.vue'
+import ETextField from '@/components/form/base/ETextField.vue'
+import ETimePicker from '@/components/form/base/ETimePicker.vue'
+
+/*
```
Comment (BacLuc) [inline frontend/src/plugins/formBaseComponents.js]: i would delete commented code which is not needed anymore

File: frontend/vite.config.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+// vite.config.js
```
Comment (BacLuc) [inline frontend/vite.config.js]: i would delete this comment, equals the filename

File: frontend/vite.config.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    ]
+  },
+  css: {
+    preprocessorOptions: {
+      scss: {
+        additionalData: [
+          '@import "./node_modules/vuetify/src/styles/styles.sass";', // original default variables from vuetify
+          '@import "@/scss/variables.scss";', // vuetify variable overrides
+          ''
+        ].join('\r\n')
```
Comment (BacLuc) [inline frontend/vite.config.js]: shouldn't \n be enough?

File: frontend/vite.config.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      },
+      sass: {
+        additionalData: [
+          '@import "./node_modules/vuetify/src/styles/styles.sass"', // original default variables from vuetify
+          '@import "@/scss/variables.scss"', // vuetify variable overrides
+          ''
+        ].join('\r\n')
```
Comment (BacLuc) [inline frontend/vite.config.js]: shouldn't \n be enough?

**Pull request merged**